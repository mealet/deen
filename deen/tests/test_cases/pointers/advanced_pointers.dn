struct Foo {
  value: i32
}

fn main() i32 {
  // pointer to struct field
  let instance = Foo { .value = 10 };
  let ptr = &instance.value;

  println!("{} | {}", *ptr, instance.value);

  *ptr = 15;

  println!("{} | {}", *ptr, instance.value);

  // pointer to array field
  let arr = [1, 2, 3];
  let ptr = &arr[0];

  println!("{} | {}", *ptr, arr[0]);

  *ptr = 133;

  println!("{} | {}", *ptr, arr[0]);

  return 0;
}
