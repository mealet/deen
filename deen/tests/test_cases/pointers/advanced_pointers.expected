10 | 10
15 | 15
1 | 1
133 | 133
